src/MetaDataIterator.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Nassau\RegistryCompiler;
+
+class MetaDataIterator extends \ArrayObject
+{
+	private $metadata;
+
+	public function __construct()
+	{
+		$this->metadata = new \SplObjectStorage();
+	}
+
+	public function offsetSet($index, $value, $metadata = null)
+	{
+		parent::offsetSet($index, $value);
+		$this->metadata->offsetSet($value, $metadata);
+	}
+
+	public function offsetUnset($index)
+	{
+		if ($this->offsetExists($index)) {
+			$this->metadata->offsetUnset($this->offsetGet($index));
```

Comment:
`\SplObjectStorage` maps objects to data, [doesn’t use keys](http://php.net/manual/en/splobjectstorage.offsetunset.php). This isn’t supposed to be used anyway.